test/acceptance/spec/06-link-checker.js

```diff
@@ -3,60 +3,53 @@
 const blc = require('broken-link-checker');
 const { url } = require('../config');
 
-const report = bl => `Broken links found:\n${Object.keys(bl).map(t => `\t- target: ${t}, ${bl[t]}`).join('\n')}\n`;
-const link   = context => l => {
-  if (l.broken && !context.broken[l.url.resolved]) {
-    context.broken[l.url.resolved] = `found on page: ${l.base.resolved}, with selector: "${l.html.selector}"`;
-  }
-};
-const checker = (followLinks, context, done) =>
-  new blc[followLinks ? 'SiteChecker' : 'HtmlChecker']({
-    filterLevel: 3,
-    excludedKeywords: ['*/oauth/logout']
-  }, followLinks ? {
-    link: link(context),
-    site: (e, url) => e && done(new Error(`${e.message} - code: ${e.code || e.status || e.statusCode} ${url}`)),
-    end: done
-  } : {
-    link: link(context),
-    complete: done
+const checkLinks = isErrorPage => page => describe(page, function() {
+  before('run check', function(done) {
+    this.broken = {};
+
+    const fullUrl = url + page;
+    const Checker = isErrorPage ? blc.HtmlChecker : blc.SiteChecker;
+    const checker = new Checker({
+      filterLevel: 3,
+      excludedKeywords: ['*/oauth/logout']
+    }, {
+      link: l => {
+        if (l.broken && !this.broken[l.url.resolved]) {
+          this.broken[l.url.resolved] = `found on page: ${l.base.resolved}, with selector: "${l.html.selector}"`;
+        }
+      },
+      site: (e, url2) => e && done(new Error(`${e.message} - code: ${e.code || e.status || e.statusCode} ${url2}`)),
+      end: done,
+      complete: done
+    });
+
+    isErrorPage ? checker.scan(browser.url(fullUrl).getHTML('html'), url) : checker.enqueue(fullUrl);
   });
 
-describe('Check for broken links', () => {
+  it('should have no broken links', function() {
+    const broken = Object.keys(this.broken).map(t => `\t- target: ${t}, ${this.broken[t]}`);
+    const report = 'Broken links found:\n' + broken.join('\n') + '\n';
```

Comment:
```javascript
    const report = `Broken links found:\n${broken.join('\n')}\n`;
```

Reply:
Ah yes. I knew this was possible but I felt that having logic inside the string, beyond simple variable substitution, made it more difficult for people to parse.

Would be good to know yours and @easternbloc's thoughts on this.

Reply:
I suppose it's a very minor bit of logic in this case.